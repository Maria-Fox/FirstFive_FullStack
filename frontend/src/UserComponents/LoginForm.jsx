import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import AlertNotification from "../Common/AlertNotifications";
import {
  Form,
  FormGroup,
  Label,
  Input,
  Card,
  CardBody,
  Button,
} from "reactstrap";

const LoginForm = ({ authenticateuser }) => {

  // ***************************************************************

  let navigate = useNavigate();

  let initial_state = {
    "username": "",
    "password": ""
  };

  let [formData, setFormData] = useState(initial_state);
  let [formErrors, setFormErrors] = useState(null);

  // ***************************************************************


  let handleChange = (e) => {
    let { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  };

  // ***************************************************************

  let handleSubmit = async (e) => {
      e.preventDefault();
      let response = await authenticateuser(formData);
      
      if (response.success) {
        navigate(`/configure`);
      } else {
        setFormErrors(response.errors);
        return;
      };
  };

  // ***************************************************************


  return (
    <div className="container" >

      <h2 className="mb-3 text-white mt-5">Login</h2>
      <Card>

        {formErrors ? <AlertNotification messages={formErrors} /> : null}

        <CardBody>
          <Form  >
            <FormGroup>
              <Label for="username" className="mt-3">
                Username
              </Label>
              <Input
                id="username"
                name="username"
                type="text"
                onChange={handleChange}
              />

              <Label for="password" className="mt-3">
                Password
              </Label>
              <Input
                id="password"
                name="password"
                type="password"
                autoComplete="on"
                onChange={handleChange}
              />
            </FormGroup>


            <Button type="submit" onClick={handleSubmit}>Submit</Button>
          </Form>
        </CardBody>
      </Card>
    </div >
  )
};

export default LoginForm;