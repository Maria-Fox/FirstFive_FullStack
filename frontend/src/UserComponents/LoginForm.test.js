import React from "react";
import { render } from "@testing-library/react";
import LoginForm from "./LoginForm";
import { MemoryRouter } from "react-router";

// smoke
test("Should render LoginForm", function () {
  render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>);
});



// snapshot test. The first time you run this will take a snapshot, then re-run to compare the outcomes.
test("Creates a LoginForm snapshot", function () {

  // as fragment is a method destructured from the return obj. Puts return item in a div and we can test what's in the fragment.
  const { asFragment } = render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});