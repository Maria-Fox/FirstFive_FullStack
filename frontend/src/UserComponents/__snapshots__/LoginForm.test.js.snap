// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a LoginForm snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2
      class="mb-3 text-white mt-5"
    >
      Login
    </h2>
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <form
          class=""
        >
          <div
            class="mb-3"
          >
            <label
              class="mt-3 form-label"
              for="username"
            >
              Username
            </label>
            <input
              class="form-control"
              id="username"
              name="username"
              type="text"
            />
            <label
              class="mt-3 form-label"
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="on"
              class="form-control"
              id="password"
              name="password"
              type="password"
            />
          </div>
          <button
            class="btn btn-secondary"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
